Packs/AzureRiskyUsers/Integrations/AzureRiskyUsers/AzureRiskyUsers.py

```diff
@@ -223,62 +225,86 @@ def get_skip_token(next_link: str | None, outputs_prefix: str, outputs_key_field
         return parse_qs(parsed_url.query)['$skiptoken'][0]
 
 
-def risky_users_list_command(client: Client, args: dict[str, str]) -> CommandResults:
+def pages_puller(client: Client, response: dict[str, Any], limit: int = 1) -> tuple[list, str | None]:
     """
-    List all risky users.
-    Args:
-        client (Client): Azure Risky Users API client.
-        args (dict): Arguments for API call.
-    Returns:
-        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    Retrieves a limited number of pages by repeatedly making requests to the API using the nextLink URL
+    until it has reached the specified limit or there are no more pages to retrieve,
+    :param response: response body, contains collection of chat/message objects
+    :param limit: the requested limit
+    :return: tuple of the limited response_data and the last nextLink URL.
     """
-    page = arg_to_number(args.get('page')) or 1
-    limit = arg_to_number(args.get('limit')) or 50
-    risk_state = args.get('risk_state')
-    risk_level = args.get('risk_level')
-    skip_token: CommandResults | str | None = None
 
-    if page > 1:
-        offset = limit * (page - 1)
+    response_data = response.get('value', [])
+    while (next_link := response.get('@odata.nextLink')) and len(response_data) < limit:
+        demisto.debug(f"Using response {next_link=}")
+        response = client.risky_users_list_request(None, None, limit, next_link)
+        response_data.extend(response.get('value', []))
+    demisto.debug(f'The limited response contains: {len(response_data[:limit])}')
+    return response_data[:limit], next_link
```

Comment:
Unless there are only 90 results, the request cannot bring 90 items. If the user requests 100 items, if there are only 90 results next_token won't be available. So the described situation is not possible, but your comment makes me think about this edge case:
1. the user requested 700 results, we are limited to 500 on a page.
2. we will bring the first page with 500 results and the second page with 500 results.
3. the next_token will point to 1001 results instead of 701 results.